epub_translator/tools/xml_like.py

```
@@ -0,0 +1,159 @@
+_FIRST_ELEMENT_PATTERN = re.compile(r'<(?![?!])[a-zA-Z]')
+_NAMESPACE_IN_TAG = re.compile(r'\{([^}]+)\}')
+
+
+class XMLLikeNode:
+    def __init__(self, file: IO[bytes]) -> None:
+        raw_content = file.read()
+        self._encoding: str = _detect_encoding(raw_content)
+        content = raw_content.decode(self._encoding)
+        self._header, xml_content = _extract_header(content)
+        try:
+            self.element = fromstring(xml_content)
```

Ruff (0.14.10)

Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)